docs/reference/search/near-real-time.asciidoc

```diff
@@ -0,0 +1,21 @@
+[[near-real-time]]
+== Near real-time search
+The overview of <<documents-indices,documents and indices>> indicates that when a document is stored in {es}, it is indexed and fully searchable in _near real-time_--within 1 second. What defines near real-time search?
+
+Lucene, the Java libraries on which {es} is based, introduced the concept of per-segment search. A _segment_ is similar to an inverted index, but the word _index_ in Lucene means "a collection of segments plus a commit point". After a commit, a new segment is added to the commit point and the buffer is cleared.
+
+Lucene allows new segments to be written and opened, making the documents they contain visible to search ​without performing a full commit. This is a much lighter process than a commit to disk, and can be done frequently without degrading performance.
+
+In {es}, this lightweight process of writing and opening a new segment is called a _refresh_. A refresh makes all operations performed on an index since the last refresh available for search. By default, every shard is refreshed automatically once per second. This is why we say that {es} has _near_ real-time search: document changes are not visible to search immediately, but will become visible within 1 second.
```

Comment:
Could we expand this to spell out all the different ways that a refresh might happen?

- wait for the refresh interval
- use the `?refresh` option
- perform an explicit `POST _refresh`

Also let's not call it "lightweight" as an absolute term. It's certainly lighter than a full commit, but it's not free and you can cause performance issues by doing it too much.

Reply:
Adding a list to explicitly mention those options, and removing "lightweight".